features/step_definitions/administration.rb

```diff
@@ -1,99 +1,110 @@
+  fill_in :admin_user_password, with: "password" 
 end
 
 Then(/^I see the admin panel$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content "Signed in successfully" 
 end
 
 Given(/^I do not have an admin account$/) do
-  pending # express the regexp above with the code you wish you had
+  AdminUser.delete_all
 end
 
 Then(/^I see a flash notification that tell me that my email does not exist in the system$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content "Invalid email or password" 
 end
 
 Given(/^I am logged into the admin panel$/) do
-  pending # express the regexp above with the code you wish you had
-end
-
-Given(/^I am logged into the site$/) do
-  pending # express the regexp above with the code you wish you had
+  @admin = AdminUser.create(email: "admin@example.com", password: "password", password_confirmation: "password")
+  visit("/admin")
+  fill_in :admin_user_email, with: "admin@example.com"
+  fill_in :admin_user_password, with: "password"
+  click_button("Login") 
 end
 
 When(/^I visit the admin books url$/) do
-  pending # express the regexp above with the code you wish you had
+  visit("/admin/books") 
 end
 
 When(/^I enter the title "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+  fill_in :book_title, with: arg1 
 end
 
-When(/^I enter the price "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+When(/^I enter the price "(.*?)"$/) do |price_cents|
+  fill_in :book_price_cents, with: price_cents 
 end
 
 When(/^I enter the published date "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+  select_date "2015,August,10", :from => "Published date"
 end
 
 When(/^I enter the author "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+  fill_in :book_author, with: "Some Person" 
 end
 
-Then(/^I see the book "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+Then(/^I see the book "(.*?)"$/) do |title|
+   expect(page).to have_content(title)
 end
 
 When(/^I visit the public book index$/) do
-  pending # express the regexp above with the code you wish you had
+  visit("/") 
 end
 
 Then(/^I see the book published date "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content "2015-08-10" 
+end
+
+Then(/^I see the book author "(.*?)"$/) do |author|
+  expect(page).to have_content "Some Person" 
+end
+
+Given(/^there is a book named "(.*?)"$/) do |title|
+  @book = Book.create(title: title ) 
 end
 
-Then(/^I see the book author "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+Given(/^the price of the book is "(.*?)"$/) do |price_cents|
+  @book.price_cents = price_cents 
```

Comment:
Watch out here. The price is not given to you in cents, but rather in dollars. Make sure you convert it.